src/agent/agent.c

```diff
@@ -369,44 +370,61 @@ bool agent_set_name(Agent *agent, const char *name) {
 }
 
 bool agent_parse_config(Agent *agent, const char *configfile) {
-        _cleanup_config_ config *config = NULL;
-        topic *topic = NULL;
-        const char *name = NULL, *host = NULL, *port = NULL;
+        int result = 0;
 
-        config = parsing_ini_file(configfile);
-        if (config == NULL) {
+        result = config_initialize();
+        if (result != 0) {
+                fprintf(stderr, "Error initializing configuration, error code '%d'.", result);
```

Comment:
```suggestion
                fprintf(stderr, "Error initializing configuration: '%s'.", strerror(-result));
```
With `strerror` we get the string for the code, e.g. `Out of memory`

Reply:
Done